fontforge/python.c

```diff
@@ -3867,27 +3982,36 @@ return( ret );
 
 static int PyFFLayer_IndexAssign( PyObject *self, Py_ssize_t pos, PyObject *val ) {
     PyFF_Layer *layer = (PyFF_Layer *) self;
-    PyFF_Contour *contour;
-    PyFF_Contour *old;
+    PyFF_Contour *old, *contour;
+    int i;
 
-    if ( !PyType_IsSubtype(&PyFF_ContourType, Py_TYPE(val)) ) {
+    if ( val!=NULL && !PyType_IsSubtype(&PyFF_ContourType, Py_TYPE(val)) ) {
 	PyErr_Format(PyExc_TypeError, "Value must be a (FontForge) Contour");
-return( -1 );
+	return( -1 );
     }
-    contour = (PyFF_Contour *) val;
     if ( pos<0 || pos>=layer->cntr_cnt ) {
 	PyErr_Format(PyExc_TypeError, "Index out of bounds");
-return( -1 );
-    }
-    if ( contour->is_quadratic!=layer->is_quadratic ) {
-	PyErr_Format(PyExc_TypeError, "Replacement contour must have the same order as the layer");
-return( -1 );
+	return( -1 );
     }
 
     old = layer->contours[pos];
-    layer->contours[pos] = contour;
+
+    if ( val==NULL) {
+	for ( i=pos; i<=layer->cntr_cnt-1; ++i )
+	    layer->contours[i] = layer->contours[i+1];
```

Comment:
Same comment here re oob acess on the contours

Reply:
It looks okay now. I assume that this was resolved?